client/signup/steps/email-signup-form/index.jsx

```diff
@@ -102,10 +102,14 @@ export default React.createClass( {
 	},
 
 	render() {
+		this.props.signupDependencies.jetpackRedirect = this.props.queryObject.jetpack_redirect;
```

Comment:
I think the correct way to set the dependencies is using:

`SignupActions.submitSignupStep` - it's the third parameter. You should be able to pass this in as part of the `submitSignupStep` call in this step.